modules/nf-core/president/tests/main.nf.test

```diff
@@ -0,0 +1,100 @@
+import java.text.SimpleDateFormat
+
+nextflow_process {
+
+    name "Test Process PRESIDENT"
+    script "../main.nf"
+    process "PRESIDENT"
+
+    tag "modules"
+    tag "modules_nfcore"
+    tag "president"
+
+    test("president - nocompress") {
+
+        when {
+            process {
+                """             
+                input[0] = [[],file(params.test_data['sarscov2']['genome']['all_sites_fas'], checkIfExists: true)]
+                input[1] = file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
+                input[2] = []
+                """
+
+            }
+        }
+
+        then {
+            def firstReportEntry = path(process.out.report.get(0).get(1)).readLines()[1].tokenize("\t")
+            def firstReportAssert = ['sample1', 'all_sites.fas', 'genome.fasta', 'False', 'False', 'False', 'False', 'False', 'True', 'True', 'False', '27627', '2276', '0', '2276', '29903', '29829', '473', '0']
+            def date = new Date()
+            def sdf = new SimpleDateFormat("yyyy-MM-dd")
+            firstReportAssert.add(sdf.format(date))
+
+            assertAll(
+                {assert process.success},
+                {assert process.out.log != null},
+                {assert process.out.log.size() == 1},
+                {assert process.out.log.get(0).get(1) ==~ ".*/president_logger.log"},
+                {assert process.out.report != null},
+                {assert process.out.report.size() == 1},
+                {assert process.out.report.get(0).get(1) ==~ ".*/report.tsv"},
+                {assert firstReportEntry == firstReportAssert},
+                {assert process.out.fasta != null},
+                {assert process.out.fasta.size() == 1},
+                {assert process.out.fasta.get(0).get(1).size() == 2},
+                {assert process.out.fasta.get(0).get(1).get(0) ==~ ".*/invalid.fasta"},
+                {assert path(process.out.fasta.get(0).get(1).get(0)).readLines().size() == 2},
+                {assert path(process.out.fasta.get(0).get(1).get(0)).md5 == "5e16f45d529bac4c6d757a84802ac772"},
+                {assert process.out.fasta.get(0).get(1).get(1) ==~ ".*/valid.fasta"},
+                {assert path(process.out.fasta.get(0).get(1).get(1)).readLines().size() == 6},
+                {assert path(process.out.fasta.get(0).get(1).get(1)).md5 == "16031b5d95fff7ff01b60615728df4ab"}
+            )
+        }
+    }
+
+    test("president - compress") {
+
+        when {
+            process {
+                """             
+                input[0] = [[],file(params.test_data['sarscov2']['genome']['all_sites_fas'], checkIfExists: true)]
+                input[1] = file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
```

Comment:
While we're here, I think you should change the file paths to match #5281

Reply:
ok, did it :)

Reply:
Thanks, I think you can merge now (if you want, but I assume you do)